misk-jobqueue/src/testFixtures/kotlin/misk/jobqueue/FakeJobQueue.kt

```diff
@@ -268,13 +290,34 @@ data class FakeJob(
   var deadLettered: Boolean = false
     internal set
 
+  var backoffDelayedTime: Instant? = null
+    internal set
+
+  var delayDuration: Long? = null
+    internal set
```

Comment:
Instead of using these two tracking vars, what about:
1. Add `var delayedForBackoff: Boolean = false internal set`
2. In `backOffDelay()`, set `delayedForBackoff = true` and set `deliveryDelay` to the new delivery time
3. In `FakeJobQueue` put the job into `jobsToQueueBack` if `delayedForBackoff == true`
4. In `FakeJobQueue`, set `delayedForBackoff = false` on each job after re-enqueueing it

Reply:
oh yes this sounds much NICER!
updated the code (had to keep the delayDuration as we needed something to keep on increasing)